src/main/java/com/st/ims/utility/AppUtility.java

```diff
@@ -3,72 +3,133 @@
 			return createdDate.plusMonths(2);
 		}
 		return createdDate;
-	
-		
+
 	}
 
-	/* This method will return alphanumeric invoice number. */
-	
+	/* This method will return alphanumeric order number. */
+
 	private static String getInvoiceNumber() throws NoSuchAlgorithmException {
 		Random rand;
 		int leftLimit = 48; // numeral '0'
 		int rightLimit = 122; // letter 'z'
 		int targetStringLength = 10;
-        rand = SecureRandom.getInstanceStrong();
- 
-		return rand.ints(leftLimit, rightLimit + 1)
-		  .filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
-		  .limit(targetStringLength)
-		  .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
-		  .toString();
-		   
-	   
+		rand = SecureRandom.getInstanceStrong();
+
+		return rand.ints(leftLimit, rightLimit + 1).filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
+				.limit(targetStringLength)
+				.collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append).toString();
+
 	}
-	
+
 	/*
-	 * This method will take invoice amount and commission percentage and will return
+	 * This method will take order amount and commission percentage and will return
 	 * commission amount.
 	 */
-	private static double getCommissionAmount(double invoiceAmount , double commissionPercentage) {
-		return invoiceAmount*(commissionPercentage/100);
+	private static double getCommissionAmount(double invoiceAmount, double commissionPercentage) {
+		return invoiceAmount * (commissionPercentage / 100);
 	}
-	
-	
+
 	public static InvoiceDetails setDefaultValues(InvoiceDetails invoiceDetail) {
 		try {
 			Invoice invoice = invoiceDetail.getInvoice();
 			invoice.setDueDate(getDueDate(invoice.getCreateDate()));
-			invoice.setCommissionAmount(getCommissionAmount(invoice.getInvoiceAmount(),invoice.getCommissionPercentage()));
+			invoice.setCommissionAmount(
+					getCommissionAmount(invoice.getInvoiceAmount(), invoice.getCommissionPercentage()));
 			invoice.setInvoiceNumber(getInvoiceNumber());
 			invoiceDetail.setInvoice(invoice);
 			return invoiceDetail;
 		} catch (NoSuchAlgorithmException e) {
-			e.printStackTrace();
+			logger.error("Exception while configuring dueDate,invoiceNumber and commissionAmount.", e);
 		}
 		return invoiceDetail;
-		
-		
+
+	}
+
+	public static Order setDefaultValues(Order order) {
+		try {
+			order.setDueDate(getDueDate(order.getCreateDate()));
+			order.setCommissionAmount(getCommissionAmount(order.getOrderAmount(), order.getCommissionPercentage()));
+			order.setOrderNumber(getInvoiceNumber());
+			return order;
+		} catch (NoSuchAlgorithmException e) {
+			logger.error("Exception while configuring dueDate,orderNumber and commissionAmount.", e);
+		}
+		return null;
+	}
+	
+	/* This method will return OrderDetails List with products. */
+	public static List<OrderDetails> getOrderDetailsList(Order order, List<OrderProducts> orderProductList){
+		try {
+			List<OrderDetails> orderDetailsList = new ArrayList<>();
```

Comment:
should decleare with null

Reply:
Compiler doesn,t allows to be null at this location.